src/audio_context.rs

```diff
@@ -602,3 +613,39 @@ fn create_media_stream_source(ctx: CallContext) -> Result<JsObject> {
 
     ctor.new_instance(&[js_this, options])
 }
+
+// ----------------------------------------------------
+// Private Event Target initialization
+// ----------------------------------------------------
+#[js_function]
+fn init_event_target(ctx: CallContext) -> Result<JsUndefined> {
+    let js_this = ctx.this_unchecked::<JsObject>();
+    let napi_obj = ctx.env.unwrap::<NapiAudioContext>(&js_this)?;
+    let context = napi_obj.0.clone();
+
+    let dispatch_event_symbol = ctx.env.symbol_for("napiDispatchEvent").unwrap();
+    let js_func = js_this.get_property(dispatch_event_symbol).unwrap();
+
+    let tsfn =
+        ctx.env
+            .create_threadsafe_function(&js_func, 0, |ctx: ThreadSafeCallContext<Event>| {
+                let event_type = ctx.env.create_string(ctx.value.type_)?;
+                Ok(vec![event_type])
+            })?;
+
+    let context_clone = context.clone();
+    context.set_onstatechange(move |e| {
+        tsfn.call(Ok(e), ThreadsafeFunctionCallMode::Blocking);
+
+        if context_clone.state() == AudioContextState::Closed {
+            // We need to clean things around so that the js object can be garbage collected.
+            // But we also need to wait so that the previous tsfn.call is executed,
+            // this is not clean, but don't see how to implement that properly right now.
+            std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
I hope this hack is only needed for the onStateChange event? Or also for other types?

Reply:
Hum no it seems to be required for other types as well, need to investigate further